generated/nidmm/session.py

```diff
@@ -562,8 +537,12 @@ def __setattr__(self, key, value):
         object.__setattr__(self, key, value)
 
     def __getitem__(self, key):
-        rep_caps = []
-        raise TypeError("'Session' object does not support indexing. You should use the applicable repeated capabilities container(s): {}".format(', '.join(rep_caps)))
+        rep_cap_help_text = ''
+        raise TypeError("'Session' object does not support indexing." + rep_cap_help_text)
+
+    def _set_use_locking(self, use_locking):
+        '''Allow runtime disabling of session locking'''
```

Comment:
Is this meant to be used internally or by users?  The `_` suggests internal / not something that can be counted on.

If it is meant to be used by users, what about the problem of someone toggling this while the lock is held?

Reply:
Since it's a workaround to an (obscure?) bug, I would not want this to not have a `_`.

But my actual preference for working around #906 would be to add a small example to the issue that teaches customers how to monkey patch their session with stub `[lock](https://nimi-python.readthedocs.io/en/1.1.0/nidmm/class.html#nidmm.Session.lock)` and `[unlock](https://nimi-python.readthedocs.io/en/1.1.0/nidmm/class.html#nidmm.Session.unlock)` implementations.

Reply:
I'd rather not encourage people to monkey patch the API.

Reply:
I put `_` because I don't plan on documenting it and I don't expect the vast majority of customers to need this. It is a workaround for a very specific issue in the driver.

I updated the docstring to be more specific when this should be called and a warning not to change it while a function call is in progress. Though if they are able to do that, then it means they don't need to set this anyway.

I also don't think we should be telling customers to monkey patch the API.

Reply:
> I'd rather not encourage people to monkey patch the API.
 
> I also don't think we should be telling customers to monkey patch the API.

I agree, but I think that's better than adding code to the API itself to workaround a bug in the drivers. Monkey patching has a lot of code smell, as it should.

Reply:
> I put _ because I don't plan on documenting it and I don't expect the vast majority of customers to need this. It is a workaround for a very specific issue in the driver.

We should have a clear message / policy when it comes to `_` functions.  Examples
- `_` are "here be dragons" but we will not break someone using it without breaking semver
- `_` is an implementation detail and we do not guarantee its behavior from release to release. Use at your own risk (and pin your version)

Reply:
> I agree, but I think that's better than adding code to the API itself to workaround a bug in the drivers. Monkey patching has a lot of code smell, as it should.

I'm fine with the workaround being listed in an issue.  Listing it in the docs is what I'd be concerned about.  That adds a level of officialness to it in my mind.

Reply:
> I'm fine with the workaround being listed in an issue.

Yep, that's what I meant with

> to add a small example to the issue

Should have been clearer.

Reply:
> We should have a clear message / policy when it comes to _ functions. Examples

My understanding is that `_` means the second one:

> _ is an implementation detail and we do not guarantee its behavior from release to release. Use at your own risk (and pin your version)
 
I can make that clearer somewhere in the wiki if you think I should.

Reply:
I removed the runtime locking disabling.